pkg/controller/registry/registry_controller_test.go

```diff
@@ -18,70 +18,290 @@
+	}
+
+	cleanupJob(t, c, jobKey)
+
+	cleanupJob(t, c, checkJobKey)
+}
+
+func createCertificateCheckingJob(t *testing.T, c client.Client, regName string) (*batchv1.Job, error) {
+
+	job := &batchv1.Job{}
+
+	registry := &kubicv1beta1.Registry{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: regName}, registry)
+	if err != nil {
+		return job, err
+	}
+
+	registryAddress := kubicutil.SafeID(registry.Spec.HostPort)
+	secret, err := registry.GetCertificateSecret(c)
+	if err != nil {
+		return job, err
+	}
+
+	nodes, err := getAllNodes(c)
+	if err != nil {
+		return job, err
+	}
+	numNodes := len(nodes)
+
+	dockerDstDir := filepath.Join(dockerCertsDir, registry.Spec.HostPort)
+	podmanDstDir := filepath.Join(podmanCertsDir, registry.Spec.HostPort)
+
+	cmdTemplate := "if [[ ! -f '%s/ca.crt' ]]; then echo 'cert not found at %s'; exit 1; fi "
+	commands := []string{
+		fmt.Sprintf(cmdTemplate, dockerDstDir, dockerDstDir),
+		fmt.Sprintf(cmdTemplate, podmanDstDir, podmanDstDir),
+	}
+
+	job, err = getRunnerJobWithSecrets(&runnerWithSecrets{
+		Commands:     []string{strings.Join(commands, " ; ")},
+		JobName:      checkJobKey.Name,
+		NumNodes:     int32(numNodes),
+		JobNamespace: metav1.NamespaceSystem,
+		Secrets:      map[string]*corev1.Secret{},
+		Labels: map[string]string{
+			jobInstallLabelHostPort: registryAddress,
+			jobInstallLabelHash:     getSecretHash(secret),
+		},
+		HostPaths: []string{
+			"/etc/docker",
+			"/etc/containers",
+		},
+		AntiAffinity: map[string]string{
+			jobInstallLabelHostPort: registryAddress,
+		},
+	})
+
+	return job, err
+}
+
+func waitRegistryCreated(c client.Client, registryName string, timeout time.Duration) error {
+
+	registry := &kubicv1beta1.Registry{}
+	var err error
+	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		err = c.Get(context.TODO(), types.NamespacedName{Name: registryName}, registry)
+
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+
+		return true, nil
+	})
+	return err
+}
+
+func waitRegistryUpdates(c client.Client, registryName string, expected int32, timeout time.Duration) (int32, error) {
+
+	registry := &kubicv1beta1.Registry{}
+	numNodes := int32(-1)
+	var err error
+	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		err = c.Get(context.TODO(), types.NamespacedName{Name: registryName}, registry)
+		if err == nil {
+			numNodes = int32(registry.Status.Certificate.NumNodes)
+		} else if !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		err = nil
+		return (numNodes == expected), nil
+	})
+	return numNodes, err
+}
+
+func waitJobCompletations(c client.Client, jobName string, expected int32, timeout time.Duration) (int32, error) {
+
+	job := &batchv1.Job{}
+	numNodes := int32(0)
+	var err error
+	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		err = c.Get(context.TODO(), types.NamespacedName{Name: jobName, Namespace: metav1.NamespaceSystem}, job)
```

Comment:
```golang
err = c.Get(context.TODO(), types.NamespacedName{Name: jobName, Namespace: metav1.NamespaceSystem}, job)
		if err != nil && !apierrors.IsNotFound(err) {
			return false, err
		}
	numNodes = job.Status.Succeeded
```

Reply:
i think we could simplify like this, removing if/else

Reply:
The code was originally like that, but if `apierrors.IsNotFound(err)` is true then accessing job.Status.Succeeded is not safe.

Reply:
ok we can refactor later then .